### PR TITLE
DOC: update the isna, isnull, notna and notnull docstring

### PR DESCRIPTION
**Pull request by Cheukting**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
The validation script did not work on this docstring

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

The validation script did not work on this docstring

### CODE REVIEW

File: pandas/core/dtypes/missing.py [hunk 1]
```diff
@@ -29,7 +29,12 @@
 
 
 def isna(obj):
-    """Detect missing values (NaN in numeric arrays, None/NaN in object arrays)
+    """Detect missing values for an array
```
Comment (TomAugspurger) [inline pandas/core/dtypes/missing.py]: End with a `.`

File: pandas/core/dtypes/missing.py [hunk 2]
```diff
@@ -46,6 +51,18 @@ def isna(obj):
     --------
     pandas.notna: boolean inverse of pandas.isna
     pandas.isnull: alias of isna
+
```
Comment (TomAugspurger) [inline pandas/core/dtypes/missing.py]: Add Series.isna, DataFrame.isna, and Index.isna.

File: pandas/core/dtypes/missing.py [hunk 3]
```diff
@@ -215,6 +237,18 @@ def notna(obj):
     --------
     pandas.isna : boolean inverse of pandas.notna
     pandas.notnull : alias of notna
+
```
Comment (TomAugspurger) [inline pandas/core/dtypes/missing.py]: Add Series.notna, DataFrame.notna, and Index.notna

File: pandas/core/dtypes/missing.py [hunk 4]
```diff
@@ -200,6 +217,11 @@ def notna(obj):
     """Replacement for numpy.isfinite / -numpy.isnan which is suitable for use
```
Comment (TomAugspurger) [inline pandas/core/dtypes/missing.py]: Can you shorten this to a single line? Can be similar to you docstring for `isna`, just flipped.

**Comment by TomAugspurger** [comment]
In the Parameters section, can you change `arr` to `obj` to match the function siganture.

File: pandas/core/dtypes/missing.py [hunk 5]
```diff
@@ -29,23 +29,53 @@
 
 
 def isna(obj):
-    """Detect missing values (NaN in numeric arrays, None/NaN in object arrays)
+    """Detect missing values for an array-like object.
+
+    This function takes an array-like object, for each element, if it is
+    a missing value (`NaN` in numeric arrays, `None`/`NaN` in object arrays)
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: NaT in datetimelike

File: pandas/core/dtypes/missing.py [hunk 6]
```diff
@@ -29,23 +29,53 @@
...
+    DataFrame.isna
+    Index.isna
+
+    Examples
     --------
-    pandas.notna: boolean inverse of pandas.isna
-    pandas.isnull: alias of isna
+    >>> pd.isna('dog')
+    False
+    >>> pd.isna(np.nan)
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: an you add blank lines between cases

File: pandas/core/dtypes/missing.py [hunk 7]
```diff
@@ -29,23 +29,53 @@
...
+    True
+    >>> array
+    array([[ 1., nan,  3.],
+           [ 4.,  5., nan]])
+    >>> pd.isna(array)
+    array([[False,  True, False],
+           [False, False,  True]])
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: an you show a datetimelike example (DatetimeIndex is good)

File: pandas/core/dtypes/missing.py [hunk 8]
```diff
@@ -197,24 +227,53 @@ def _isna_ndarraylike_old(obj):
 
 
 def notna(obj):
-    """Replacement for numpy.isfinite / -numpy.isnan which is suitable for use
-    on object arrays.
+    """Detect non-missing values for an array-like object.
+
+    This function takes an array-like object, for each element, if it is *not*
+    a missing.value (`NaN` in numeric arrays, `None`/`NaN` in object arrays)
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: same as above

File: pandas/core/dtypes/missing.py [hunk 9]
```diff
@@ -197,24 +227,53 @@ def _isna_ndarraylike_old(obj):
...
     --------
-    pandas.isna : boolean inverse of pandas.notna
-    pandas.notnull : alias of notna
+    >>> pd.notna('dog')
+    True
+    >>> pd.notna(np.nan)
+    False
+    >>> array
+    array([[ 1., nan,  3.],
+           [ 4.,  5., nan]])
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/dtypes/missing.py [hunk 10]
```diff
@@ -29,23 +29,53 @@
...
     ----------
-    arr : ndarray or object value
-        Object to check for null-ness
+    obj : array-like or object
+        Object to check for null-ness.
 
     Returns
     -------
-    isna : array-like of bool or bool
+    array-like of bool or bool
```
Comment (TomAugspurger) [inline pandas/core/dtypes/missing.py]: Could you rephrase as `bool or array-like of bool`, so that it goes small to big?

File: pandas/core/dtypes/missing.py [hunk 11]
```diff
@@ -29,23 +29,53 @@
...
         Array or bool indicating whether an object is null or if an array is
```
Comment (TomAugspurger) [inline pandas/core/dtypes/missing.py]: This wording tripped me up. Could try rephrasing it? Maybe something like, "Indicator for missing values. For a scalar, input, a scalar boolean is returned. For an array-like input, and array (of the same type?) of booleans indicating NA values element-wise is returned.

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks @Cheukting !
